numpy/lib/twodim_base.py

```diff
@@ -1010,7 +1014,10 @@ def triu_indices(n, k=0, m=None):
            [ 12,  13,  14,  -1]])
 
     """
-    return nonzero(~tri(n, m, k=k-1, dtype=bool))
+    tri_ = ~tri(n, m, k=k - 1, dtype=bool)
```

Comment:
style nit: 
```suggestion
    tri_ = not tri(n, m, k=k - 1, dtype=bool)
```

Reply:
Did this work for you? 

`not` does not seem to be elementwise so I think we'll stick to `~`:
```python
not np.tri(2, 2, k=0, dtype=bool)
Traceback (most recent call last):

  File "<ipython-input-15-170be892b154>", line 1, in <module>
    not np.tri(2, 2, k=0, dtype=bool)

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Reply:
Yeah, @mattip's suggestion is incorrect here, I assume they misread `tri_` as being a scalar.